infra/blueprint-test/pkg/tft/terraform.go

```diff
@@ -277,6 +278,15 @@ func (b *TFBlueprintTest) sensitiveOutputs(dir string) map[string]bool {
 
 // getOutputs returns all output values.
 func (b *TFBlueprintTest) getOutputs(sensitive map[string]bool) map[string]interface{} {
+	// allow only parallel reads as Terraform plugin cache isn't concurrent safe
+	if err := b.tftCacheMutex.RLock(); err != nil {
+		b.t.Fatalf("Could not acquire read lock: %v", err)
+	}
+	defer func() {
+		if err := b.tftCacheMutex.RUnlock(); err != nil {
+			b.t.Fatalf("Could not release read lock: %v", err)
+		}
+	}()
```

Comment:
nit: since we are starting to do this more, it maybe be good to pull into a func that could be added at sites like 

```
rUnlockFn := b.rLock()
defer rUnlockFn()
```

Reply:
SG - I was thinking about adding convenience functions, but didn't think to return the unlock function! :)